### PR TITLE
Use provider config options with default configs

### LINKED ISSUE #199: Use providerSettings config to pass provider options
Because we will soon support multiple providers with kantra, we want to be able to pass individual provider options in a config file. Right now, we can pass a few Java-specific options via kantra cli. If we continue down this path with other providers, we will have an overwhelming number of flag options.

### PR DESCRIPTION
**Pull request by eemcmullan**
Closes #199

### CODE REVIEW

**Review by aufi — APPROVED** [review]
LGTM

**Review by pranavgaikwad — COMMENTED** [review]
lets keep maven settings file option for now

File: cmd/analyze.go [hunk 1]
```diff
@@ -1366,6 +1352,115 @@ func (a *analyzeCommand) ConvertXML(ctx context.Context) (string, error) {
...
+		a.log.V(1).Error(err,
+			"failed to write provider config", "dir", tempDir, "file", "settings.json")
+		return err
+	}
+	return nil
+}
+
+func (a *analyzeCommand) getProviderOptions(tempDir string, provConfig []provider.Config, prov string) error {
+	// get provider options from provider settings file
+	home := os.Getenv("HOME")
```
Comment (pranavgaikwad) [inline cmd/analyze.go:1372]: does this work on windows?

Comment (eemcmullan) [reply cmd/analyze.go:1372]: I believe so, as we've had other instances of using this in other places for a while now, such as https://github.com/konveyor/kantra/blob/main/cmd/settings.go#L57 without problems. Also, the os package explains differences for windows where needed, and does not mention any for `getenv` https://pkg.go.dev/os#Getenv

Comment (shawn-hurley) [inline cmd/analyze.go:1372]: I like that on linux, we may want to consider: 

```
$XDG_CONFIG_HOME defines the base directory relative to which user-specific configuration files should be stored. If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used.
```

File: cmd/analyze.go [hunk 2]
```diff
@@ -1366,6 +1352,115 @@ func (a *analyzeCommand) ConvertXML(ctx context.Context) (string, error) {
...
+	merged := []provider.Config{}
+	seen := map[string]*provider.Config{}
+
+	// find options used for supported providers
+	for idx, conf := range defaultConf {
+		seen[conf.Name] = &defaultConf[idx]
+	}
+
+	for _, conf := range optionsConf {
+		if _, ok := seen[conf.Name]; ok {
```
Comment (shawn-hurley) [inline cmd/analyze.go:1404]: nit, for reducing the complexity, continue the for loop on `!ok` and then move the code in this block up a level

File: cmd/analyze.go [hunk 3]
```diff
@@ -1366,6 +1352,115 @@ func (a *analyzeCommand) ConvertXML(ctx context.Context) (string, error) {
...
+	}
+	for _, v := range seen {
+		merged = append(merged, *v)
+	}
+	return merged, nil
+}
+
+func (a *analyzeCommand) mergeProviderSpecificConfig(optionsConf, seenConf map[string]interface{}, tempDir string) (map[string]interface{}, error) {
+	for k, v := range optionsConf {
+		switch {
```
Comment (shawn-hurley) [inline cmd/analyze.go:1435]: is there meant to be a default case?

**Review by shawn-hurley — COMMENTED** [review]
This is looking good, I think this is going to be an excellent user experience!

File: cmd/analyze.go (hunk 1 above)
Comment (pranavgaikwad) [reply cmd/analyze.go:1372]: I meant the "HOME" variable

Comment (eemcmullan) [reply cmd/analyze.go:1372]: That does make more sense. Will update

File: cmd/analyze.go (hunk 3 above)
Comment (eemcmullan) [reply cmd/analyze.go:1435]: I'm not sure there are any empty fields we'd want to add to the config if not there

File: cmd/analyze.go (hunk 2 above)
Comment (eemcmullan) [reply cmd/analyze.go:1404]: good catch

File: cmd/analyze.go (hunk 3 above)
Comment (shawn-hurley) [reply cmd/analyze.go:1585]: I agree, resolving this one

**Review by shawn-hurley — APPROVED** [review]
besides the complexity nit LGTM

**Pull request merged**